client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -323,12 +360,81 @@ def predict_with_id(
 
+    def batch_predict(
+            self,
+            x: pd.DataFrame,
+            batch_size: int = 100,
+            max_retries: int = http_session.DEFAULT_MAX_RETRIES,
+            retry_status: Set[int] = http_session.DEFAULT_STATUS_FORCELIST,
+            backoff_factor: float = http_session.DEFAULT_BACKOFF_FACTOR,
+            prediction_id: str = None,
+    ) -> pd.DataFrame:
+        """
+        Makes a prediction using input `x`.
+
+        Parameters
+        ----------
+        x : pd.DataFrame
+            A batch of inputs for the model.
+        compress : bool, default False
+            Whether to compress the request body.
+        batch_size : int, default 100
+            The number of rows to send in each request.
+        max_retries : int, default 13
+            Maximum number of retries on status codes listed in ``retry_status``.
+        retry_status : set, default {404, 429, 500, 503, 504}
+            Set of status codes, as integers, for which retry attempts should be made.  Overwrites default value.
+            Expand the set to include more. For example, to add status code 409 to the existing set, use:
+            ``retry_status={404, 429, 500, 503, 504, 409}``
+        backoff_factor : float, default 0.3
+            A backoff factor to apply between retry attempts.  Uses standard urllib3 sleep pattern:
+            ``{backoff factor} * (2 ** ({number of total retries} - 1))`` with a maximum sleep time between requests of
+            120 seconds.
+        prediction_id: str, default None
+            A custom string to use as the ID for the prediction request.  Defaults to a randomly generated UUID.
+        Returns
+        -------
+        prediction : pd.DataFrame
+            Output returned by the deployed model for `x`.
+
+        Raises
+        ------
+        RuntimeError
+            If the deployed model encounters an error while running the prediction.
+        requests.HTTPError
+            If the server encounters an error while handing the HTTP request.
+
+        """
+        # TODO: batch prediction with id
+        # Set the retry config if it differs from current config.
+        self._session = http_session.set_retry_config(
+            self._session,
+            max_retries=max_retries,
+            status_forcelist=retry_status,
+            backoff_factor=backoff_factor,
+        )
+
+        # TODO: add compression
+        # Split into batches
+        out_df_list = []
+        batches = np.array_split(x, batch_size)
+        for batch in batches:
+            # response = self._single_batch_predict(batch, batch_size, prediction_id)
+            # out_batches.extend(_utils.body_to_json(response))
+
+            # TODO: obviously undo before merging
+            temp = pd.DataFrame.from_dict({'name': ['Alice', 'Bob', 'Charlie'],
+                                           'age': [25, 30, 35],
+                                           'city': ['New York', 'London', 'Paris']})
```

Comment:
In my last PR, @liuverta mentioned that we should prefix incomplete functions under development with an underscore such that they don't show up in the docs before they are actually usable.  Can I assume there is some means by which the publicly available docs get updated independently of a client release?  (e.g. pulled from `main` periodically)

Reply:
The only reason we really needed to do that was because we we're right about to ship a new client release.

Docs currently are updated every time we merge to `main`.